modules/utxo-lib/test/integration_local_rpc/compare.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @prettier
+ */
+import { RpcTransaction } from './generate/RpcTypes';
+import * as networks from '../../src/networks';
+import { Network } from '../../src/networkTypes';
+import { getMainnet, isZcash } from '../../src/coins';
+
+type NormalizedObject = Record<string, unknown>;
+
+export function normalizeParsedTransaction(tx, network: Network): NormalizedObject {
+  const normalizedTx: NormalizedObject = {
+    txid: tx.getId(),
+    version: tx.version,
+    hex: tx.toBuffer().toString('hex'),
+    locktime: tx.locktime,
+    size: tx.byteLength(),
+    vin: tx.ins.map((i) => {
+      const normalizedInput: NormalizedObject = {
+        scriptSig: {
+          hex: i.script.toString('hex'),
+        },
+        sequence: i.sequence,
+        txid: Buffer.from(i.hash).reverse().toString('hex'),
+        vout: i.index,
+      };
+
+      if (i.witness && i.witness.length) {
+        normalizedInput.txinwitness = i.witness.map((w) => w.toString('hex'));
+      }
+
+      return normalizedInput;
+    }),
+    vout: tx.outs.map((o, n) => {
+      return {
+        n,
+        scriptPubKey: {
+          hex: o.script.toString('hex'),
+        },
+        value: o.value,
+      };
+    }),
+  };
+
+  switch (getMainnet(network)) {
+    case networks.bitcoin:
+    case networks.bitcoingold:
+    case networks.litecoin:
+      normalizedTx.vsize = tx.virtualSize();
+      normalizedTx.weight = tx.weight();
+      break;
+    case networks.dash:
+      normalizedTx.type = tx.type;
+      if (tx.extraPayload && tx.extraPayload.length) {
+        normalizedTx.extraPayload = tx.extraPayload.toString('hex');
+        normalizedTx.extraPayloadSize = tx.extraPayload.length;
+      }
+      break;
+    case networks.zcash:
+      normalizedTx.overwintered = !!tx.overwintered;
+      normalizedTx.versiongroupid = tx.versionGroupId.toString(16);
+      normalizedTx.expiryheight = tx.expiryHeight;
+      normalizedTx.vjoinsplit = [];
+      normalizedTx.vShieldedOutput = [];
+      normalizedTx.vShieldedSpend = [];
+      normalizedTx.valueBalance = 0;
+  }
+
+  return normalizedTx;
+}
+
+export function normalizeRpcTransaction(tx: RpcTransaction, network: Network): NormalizedObject {
+  const normalizedTx: NormalizedObject = {
+    ...tx,
+    vin: tx.vin.map((v: any) => {
+      delete v.scriptSig.asm;
```

Comment:
In this case why not just `return { ....v, scriptSig: { hex: v.scriptSig.hex }};` ? In other words just returning only the keys you want, rather than deleting the keys you don't want.

Reply:
If there is some new protocol update or new coin where inputs have a new property `foo`, I want to deal with it explicitly. 

I probably want to delete/ignore it but I might also want to compare it against the parsed object.